server/src/services/test_service.ts

```diff
@@ -0,0 +1,47 @@
+import { inject, injectable, named } from "inversify";
+import { TestRepository } from "../repositories/test_repository";
+import { Role } from "../entities/role";
+import { TestResults } from "../entities/test";
+import { AuthenticationService } from "./authentication_service";
+import { AuthorizationError } from "../entities/errors/authorization_error";
+import { RequestAddress } from "../entities/request/RequestAddress";
+import { UserRepository } from "../repositories/user_repository";
+
+@injectable()
+export class TestService {
+    constructor(
+        @inject("Repository")
+        @named("TestRepository")
+        private readonly testRepository: TestRepository,
+        @inject("Repository")
+        @named("UserRepository")
+        private readonly userRepository: UserRepository,
+        @inject("Service")
+        @named("AuthenticationService")
+        private readonly authenticationService: AuthenticationService,
+    ) {}
+
+    async addTestResults(
```

Comment:
I like to name the controller and the service it calls the same function name

```suggestion
    async postTestResult(
```